app/models/period.rb

```diff
@@ -27,6 +27,15 @@ def self.quarter(value)
   def initialize(attributes = {})
     super
     @type = type.intern if type
+    @value = if @value.is_a?(String)
+      if quarter?
+        Quarter.parse(@value)
+      else
+        @value.to_date
+      end
+    else
+      @value
+    end
```

Comment:
Is there a way where we can avoid shadowing `@value`? Also the `if-else` could go in a separate method.

Reply:
Yup, I didn't really like this. I cleaned it up a bit to try to make it more clear.